Source/OxyPlot/Utilities/Timer.cs

```diff
@@ -0,0 +1,250 @@
+﻿// --------------------------------------------------------------------------------------------------------------------
+// <copyright file="Timer.cs" company="OxyPlot">
+//   Copyright (c) 2014 OxyPlot contributors
+// </copyright>
+// <summary>
+//   Provides a timer that works for all target platforms.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace OxyPlot
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Timer callback delegate.
+    /// </summary>
+    /// <param name="state">
+    /// The state.
+    /// </param>
+    public delegate void TimerCallback(object state);
+
+    /// <summary>
+    /// Timer for WinRT since WinRT only provides the DispatcherTimer which cannot be used outside the UI thread.
+    /// </summary>
+    public class Timer : IDisposable
```

Comment:
Not original code, it comes from Catel (I wrote it, have to fight all the platform differences there as well). It's MIT licensed so you can do whatever you want with it:

https://github.com/Catel/Catel/blob/develop/src/Catel.Core/Catel.Core.Shared/Threading/Timer.cs